apps/webapp/src/state/helpers.ts

```diff
@@ -110,3 +111,17 @@ export const getTxId = (tx: Transaction | PartialMessage<Transaction>) =>
   sha256Hash(tx instanceof Transaction ? tx.toBinary() : new Transaction(tx).toBinary()).then(
     inner => new TransactionId({ inner }),
   );
+
+/**
+ * @todo: The error flow between extension <-> webapp needs to be refactored a
+ * bit. Right now, if we throw a `ConnectError` with `Code.PermissionDenied` (as
+ * we do in the approver), it gets swallowed by ConnectRPC's internals and
+ * rethrown via `ConnectError.from()`.  This means that the original code is
+ * lost, although the stringified error message still contains
+ * `[permission_denied]`. So we'll (somewhat hackily) check the stringified
+ * error message for now; but in the future, we need ot get the error flow
+ * working properly so that we can actually check `e.code ===
+ * Code.PermissionDenied`.
+ */
+export const userDeniedTransaction = (e: unknown): boolean =>
```

Comment:
Realized this was only in the `send` slice, but should have also been in the `swap` and `ibc` slices, so I pulled it out into a helper.